test/conformance/api/v1/revision_timeout_test.go

```diff
@@ -100,134 +83,119 @@ func TestRevisionTimeout(t *testing.T) {
+		timeoutSeconds: 10,
+		expectedStatus: http.StatusOK,
+		initialSleep:   2,
+		sleep:          0,
+	}, {
+		name:           "when pods already exist, and it does exceed timeout seconds",
+		timeoutSeconds: 10,
+		expectedStatus: http.StatusGatewayTimeout,
+		initialSleep:   12,
+		sleep:          0,
+	}, {
+		name:           "when pods already exist, and it writes first byte before timeout",
+		timeoutSeconds: 10,
+		expectedStatus: http.StatusOK,
+		initialSleep:   0,
+		sleep:          15,
+	}}
 
-	// Not fail by not surpassing in the initial timeout, but in the overall request duration.
-	if err := sendRequest(t, clients, rev2sURL, time.Second, 3*time.Second, http.StatusOK); err != nil {
-		t.Errorf("Did not fail request with sleep 1s/3s with revision timeout 2s: %v", err)
-	}
-	if err := sendRequest(t, clients, rev5sURL, 3*time.Second, 3*time.Second, http.StatusOK); err != nil {
-		t.Errorf("Failed request with sleep 3s/3s with revision timeout 5s: %v", err)
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			names := test.ResourceNames{
+				Service: test.ObjectNameForTest(t),
+				Image:   test.Timeout,
+			}
+
+			test.CleanupOnInterrupt(func() { test.TearDown(clients, names) })
+			defer test.TearDown(clients, names)
+
+			t.Log("Creating a new Service ")
+			_, err := createService(t, clients, names, tc.timeoutSeconds)
+			if err != nil {
+				t.Fatal("Failed to create Service:", err)
+			}
+
+			t.Log("The Service will be updated with the name of the Revision once it is created")
+			revisionName, err := v1test.WaitForServiceLatestRevision(clients, names)
+			if err != nil {
+				t.Fatalf("Service %s was not updated with the new revision: %v", names.Service, err)
+			}
+			names.Revision = revisionName
+
+			t.Log("When the Service reports as Ready, everything should be ready")
+			if err := v1test.WaitForServiceState(clients.ServingClient, names.Service, v1test.IsServiceReady, "ServiceIsReady"); err != nil {
+				t.Fatalf("The Service %s was not marked as Ready to serve traffic to Revision %s: %v", names.Service, names.Revision, err)
+			}
+
+			service, err := clients.ServingClient.Services.Get(names.Service, metav1.GetOptions{})
+			if err != nil {
+				t.Fatalf("Error fetching Service %s: %v", names.Service, err)
+			}
+
+			if service.Status.URL == nil {
+				t.Fatalf("Unable to fetch URLs from service: %#v", service.Status)
+			}
+
+			serviceURL := url.URL(*service.Status.URL)
+
+			if tc.shouldScaleTo0 {
+				t.Log("Waiting to scale down to 0")
+				revision, err := clients.ServingClient.Revisions.Get(names.Revision, metav1.GetOptions{})
+				if err != nil {
+					t.Fatalf("Error fetching Service %s: %v", names.Service, err)
+				}
+
+				if err := e2e.WaitForScaleToZero(t, revisionresourcenames.Deployment(revision), clients); err != nil {
+					t.Fatal("Could not scale to zero:", err)
+				}
+			} else {
+				t.Log("Probing to force at least one pod", &serviceURL)
```

Comment:
```suggestion
				t.Log("Probing to force at least one pod", serviceURL)
```

Reply:
This change will print
`Probing to force at least one pod {http   revision-timeout-when-3816fd29558419e313e3315aca17d5f3-sjbqsxhb.serving-tests.<some_ip>.xip.io   false  }`

rather than
`Probing to force at least one pod http://revision-timeout-when-b67b736422814c3736c90c658f29bd58-hrmdywoc.serving-tests.<some_ip>.xip.io`

LMK if you still want this change.

Reply:
serviceURL.String()?

Reply:
or maybe even better, could just do the cast as `(*url.URL)(service.Status.URL)` on L169 and avoid all the `&`s entirely?